[CHNL-5370] Updating readme to include instructions on how to use `initialize` and `setToken` from RN code. (#113)
klaviyo/klaviyo-react-native-sdk: README.md

The SDK must be initialized with the short alphanumeric
[public API key](https://help.klaviyo.com/hc/en-us/articles/115005062267#difference-between-public-and-private-api-keys1)
for your Klaviyo account, also known as your Site ID.

Initialization can either be done from your app's React native or native code, and must occur before
any other SDK methods can be invoked so that Klaviyo SDK can track profiles, events and push tokens
towards your company.

The decision to initialize the SDK from the React Native or the native code depends on your app's architecture.

> Finally, the SDK initialization needs to be done just once either in your react native or native code.

Reviewer: I'd take out "finally"

Reviewer: In fact I think I'd take out this whole sentence. If anything, you could say more in the sentence above, but I think this is quite clear:

> Initialization can either be done from your app's React native or native code

> for your Klaviyo account, also known as your Site ID.
> 
> Initialization can either be done from your app's React native or native code, and must occur before
> any other SDK methods can be invoked so that Klaviyo SDK can track profiles, events and push tokens
> towards your company.
> 
> The decision to initialize the SDK from the React Native or the native code depends on your app's architecture.
> 
> Finally, the SDK initialization needs to be done just once either in your react native or native code.
> It's not required to initialize the SDK in both places.

Reviewer: ```suggestion
for your Klaviyo account, also known as your Site ID.

Initialize *must* be called prior to invoking any other SDK methods so that Klaviyo SDK can track profiles, events and push tokens toward the correct Klaviyo account. Any SDK operations invoked before initialize will be dropped, and result in a logged error.

You can call `initialize` from your app's React Native layer or from the platform-specific native code. 
This decision is dependent only upon your app's architecture. It is not required to initialize the SDK in both places!
Note: It is safe to re-initialize, e.g. if your app can connect to more than one Klaviyo account. 
```

### React Native Initialization

> > ℹ️ A caveat to initializing the SDK from React Native is that this call should happen before any another calls to Klaviyo SDKs

Reviewer 2: Hmm feels a little repetitive.

Author: Yeah it was intentional so that folks don't miss it. Would it be better if I remove this from above and keep this?

Reviewer 2: Yah I think so!

Reviewer: Agreed

Author: I removed this since the initialize comment above seemed to apply to both cases I.E RN and native initialization.

### Collecting Push Tokens

> Push tokens must be collected both in your app's react native code or in the native code. Below sections discuss both approaches, and

Reviewer: ```suggestion
Push tokens can be collected either from your app's react native code or in the native code. Below sections discuss both approaches, and
```

you are free to pick one that best suits your app's architecture. Note that doing this in one location is sufficient.

#### React Native Token Collection

> In order to collect the APNs push token in your React Native code you need to,

Reviewer 2: So just to be clear we are recommending adding a firebase library to collect APNs tokens. We think this is ok since they will probably already be using this library right?

Reviewer: ```suggestion
In order to collect the APNs push token in your React Native code you need to:
```

> 1.  Import libraries such as [`@react-native-firebase/messaging`](https://www.npmjs.com/package/@react-native-firebase/messaging) to your react native project. The below instructions are specific for `@react-native-firebase/messaging` library.

Reviewer: ```suggestion
1.  Import a library such as [`@react-native-firebase/messaging`](https://www.npmjs.com/package/@react-native-firebase/messaging) to your react native project. The below instructions are specific for `@react-native-firebase/messaging` library.
```

2.  Import Firebase iOS SDK to your iOS project. Setup instructions can be found [here](https://firebase.google.com/docs/ios/setup).

> 3.  In order for the Klaviyo SDK to handle push opens via the notification center delegate methods, we need to disable

Reviewer 2: Do we want information about push opens in the token collection section?

Reviewer: I wonder the same... spitballing here but would it make sense to just have a section  `Integrate with React Native Firebase` instead maybe

Author: > Do we want information about push opens in the token collection section?

Yeah, I was not fully convinced as well but did seems related to method swizzling topic which is discussed here. May be this can be reworded to just say that method swizzling should be disabled in order for you notification center delegates in AppDelegate to be called.

Reviewer: Oh yeah that should sound better

Author: @evan-masseau  I was trying to keep the instructions pretty loose and just suggest firebase as an option w/o recommending it.

Reviewer: Yeah that's a good point better to keep it generic than have a specific firebase section

5. Finally, in your React Native code, you can collect & set the push token as follows:

```typescript
import messaging from '@react-native-firebase/messaging';
import { Klaviyo } from 'klaviyo-react-native-sdk';
import { Platform } from 'react-native';

const fetchAndSetPushToken = async () => {
  try {
    let deviceToken: string | null = null;
    if (Platform.OS === 'android') {
      deviceToken = await messaging().getToken();
      console.log('FCM Token:', deviceToken);
    } else {
      deviceToken = await messaging().getAPNSToken();
      console.log('APNs Token:', deviceToken);
    }

    if (deviceToken != null && deviceToken.length > 0) {
      Klaviyo.setPushToken(deviceToken!);
    }
  } catch (error) {
    console.error('Error in fetchAndSetPushToken:', error);
  }
};
```

For android token collection, there isn't any additional setup required on the native side. The above code should work as is.

#### Native Token Collection

Follow the platform-specific instructions below:

- [Android](https://github.com/klaviyo/klaviyo-android-sdk#Collecting-Push-Tokens)
- [iOS](https://github.com/klaviyo/klaviyo-swift-sdk#Collecting-Push-Tokens)

#### Notification Permission

> Requesting user permission to display notifications can be managed in,

Reviewer 2: is a colon better here?

Reviewer: ```suggestion
Requesting user permission to display notifications can be managed in:
```
